src/process/mod.rs

```diff
@@ -529,20 +686,13 @@ pub struct MemoryMap {
     /// BSS (uninitialized data).
     pub inode: u64,
     pub pathname: MMapPath,
+    /// Memory mapping extension information, populated when parsing `/proc/<pid>/smaps`.
+    ///
+    /// The members will be `Default::default()` (empty/none) when the information isn't available.
+    pub extension: MMapExtension,
```

Comment:
I'm wondering if anyone will have a usecase where they want to quickly know if the `extension` data is empty/none.  Would an `Option<MMapExtension>` type be better?

No need to change anything now, just starting the brainstorming process.

Reply:
I considered that, and intended to add an `is_empty` method but forgot. Though I can't actually think of a convincing use case, only because the typical "eliding unnecessary code" use case is pretty easy to check by accessing the inner members (e.g. `map`). I chose not to add `Option` because I thought it would be more ergonomic for more general code to simply query the inner members rather than having to match and remove the `Option` layer in some way.

Reply:
I'm liking the `is_empty` solution 👍